src/libraries/Microsoft.Extensions.Logging.Console/src/StringWriter.cs

```diff
@@ -0,0 +1,182 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Text;
+using System.Collections.Concurrent;
+
+namespace Microsoft.Extensions.Logging.Console
+{
+    internal class StringWriter : TextWriter
+    {
+        private readonly StringBuilder _stringBuilder;
+        internal bool DisableColors { get; set; }
+
+        public override Encoding Encoding => Encoding.Unicode;
+
+        public StringWriter(bool disableColors = false)
+        {
+            _stringBuilder = new StringBuilder();
+            DisableColors = disableColors;
+        }
+
+        public override void Write(char value)
+        {
+            _stringBuilder.Append(value);
+        }
+
+        public bool SetBackgroundColor(ConsoleColor? background)
```

Comment:
>? [](start = 51, length = 1)

why we are using nullable types here at all? is it expected someone call with uninitialized values? this look strange to me.

Reply:
this is unrefactored code from before. ConsoleColor is actually an enum and with nullable type this basically means reset Background color to no color